src/analysis/processing/qgsalgorithmmultiparttosinglepart.cpp

```diff
@@ -66,6 +67,14 @@ QgsMultipartToSinglepartAlgorithm *QgsMultipartToSinglepartAlgorithm::createInst
   return new QgsMultipartToSinglepartAlgorithm();
 }
 
+bool QgsMultipartToSinglepartAlgorithm::supportInPlaceEdit( const QgsVectorLayer *layer ) const
+{
+  if ( ! QgsProcessingFeatureBasedAlgorithm::supportInPlaceEdit( layer ) )
+    return false;
+  // The layer is already single part!
+  return QgsWkbTypes::isMultiType( layer->wkbType() );
```

Comment:
As discussed earlier, i think we should allow this, just in case something earlier has caused uncommitted features to be multipart and they need fixing to save into a singlepart provider.